test/embedded-refract-test.js

```diff
@@ -0,0 +1,442 @@
+var _ = require('lodash');
+var expect = require('./spec-helper').expect;
+var minim = require('../lib/minim');
+
+describe('Embedded Refract', function() {
+  context('when converting from embedded refract', function() {
+    context('when given a normal object', function() {
+      var obj = {
+        foo: 'bar'
+      };
+
+      var element;
+
+      before(function() {
+        element = minim.convertFromEmbedded(obj);
+      });
+
+      it('returns the correct value', function() {
+        expect(element.toValue()).to.deep.equal(obj);
+      });
+    });
+
+    context('when given embedded refract object', function() {
+      // An object { b: 2 } with embedded refract
+      var obj = {
+        refract: {
+          element: 'object',
+          meta: {
+            id: 'test-id'
+          },
+          attributes: {
+            a: 1
+          }
+        },
+        b: 2
+      }
+
+      var refract = {
+        element: 'object',
+        meta: {
+          id: 'test-id'
+        },
+        attributes: {
+          a: 1
+        },
+        content: [
+          {
+            element: 'member',
+            meta: {},
+            attributes: {},
+            content: {
+              key: {
+                element: 'string',
+                meta: {},
+                attributes: {},
+                content: 'b'
+              },
+              value: {
+                element: 'number',
+                meta: {},
+                attributes: {},
+                content: 2
+              }
+            }
+          }
+        ]
+      };
+
+      var element;
+
+      before(function() {
+        element = minim.convertFromEmbedded(obj);
+      });
+
+      it('returns the correct value', function() {
+        expect(element.toRefract()).to.deep.equal(refract);
+      });
+    });
+
+    context('when a property is embedded refract', function() {
+      // Object { a: 1, b: 2, c: 3 } but the value of `b` is embedded refract
+      var obj = {
+        a: 1,
+        b: {
```

Comment:
@smizell I don't think that is a consistent interpretation with what you see here: https://github.com/refractproject/minim/pull/56/files#diff-d4f2450b340a251071ab34ac7d28869eR25